src/client.ts

```diff
@@ -7235,7 +7216,13 @@ export class MatrixClient extends EventEmitter {
      * @return {Promise} Resolves: an empty object {}
      * @return {module:http-api.MatrixError} Rejects: with an error response.
      */
-    public addPushRule(scope: string, kind: string, ruleId: string, body: any, callback?: Callback): Promise<{}> {
+    public addPushRule(
+        scope: string,
+        kind: PushRuleKind,
+        ruleId: RuleId | string,
```

Comment:
this actually can't be RuleId as you cannot add those given the `.` prefix - maybe `Exclude<string, RuleId>` or something